Backend/src/main/java/com/example/Backend/advice/UserControllerAdvice.java

```diff
@@ -0,0 +1,54 @@
+package com.example.Backend.advice;
+
+import com.example.Backend.exceptions.InvalidInputException;
+import com.example.Backend.exceptions.UserNotFoundException;
+import com.example.Backend.exceptions.UserRegistrationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ControllerAdvice
+public class UserControllerAdvice {
+
+    @ExceptionHandler(UserNotFoundException.class)
+    public ResponseEntity<Object> handleUserNotFoundException(
+            UserNotFoundException ex) {
+        Map<String, Object> body = new HashMap<>();
+        body.put("status", HttpStatus.NOT_FOUND.value());
+        body.put("error", "User Not Found");
+        body.put("message", ex.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(UserRegistrationException.class)
+    public ResponseEntity<Object> handleUserRegistrationException(
+            UserRegistrationException ex) {
+        Map<String, Object> body = new HashMap<>();
```

Comment:
Just like we created a request DTO. create a generic response DTO as well. Use that instead of a Map